### PR TITLE
33 refacto entity serialization

### PR DESCRIPTION
**Pull request by nLatt**
Refactoring the Entity Serialization to fit the new ECS while adding syntax modification

### CODE REVIEW

**Comment by mindoodoo** [comment]
DESCRIPTION BITTECHEUN @nLatt

File: src/server/main.cpp [hunk 1]
```diff
@@ -1,58 +1,36 @@
 #include <asio.hpp>
```
Comment (mindoodoo) [inline src/server/main.cpp:1]: Is it ok if the linting workflow has an issue with the header not found ?

File: src/shared/ECS/Components.hpp [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+struct Armor {
+    int armor = 0;
+};
+
+struct Health {
+    int health = 0;
+};
+
+struct Position {
+    float xpos = 0;
```
Comment (mindoodoo) [inline src/shared/ECS/Components.hpp:21]: camelCase ?

File: src/shared/systems/serialization.hpp [hunk 3]
```diff
@@ -0,0 +1,140 @@
...
+}
+std::string toString(Position component) {
+    std::stringstream ss;
+
+    ss << "(" << component.xpos << "," << component.ypos << ")";
+    return ss.str();
+}
+
+template <class T>
+std::string componentToString(EntityID entityId, Manager* manager) {
```
Comment (mindoodoo) [inline src/shared/systems/serialization.hpp:39]: **null_ptr** if entity does not have the specified component unhandled, this is a potential crash waiting to happen :bomb:

Comment (mindoodoo) [reply src/shared/systems/serialization.hpp:39]: Could be handled with no extra handling needed in entityToString() by returning empty string

File: src/shared/systems/serialization.hpp [hunk 4]
```diff
@@ -0,0 +1,140 @@
...
+
+    std::sregex_iterator words_begin(match.str().begin(), match.str().end(), argRegex);
+    std::sregex_iterator words_end = std::sregex_iterator();
+
+    for (std::sregex_iterator i = words_begin; i != words_end; ++i) {
+        args.push_back((*i).str());
+    }
+
+    return args;
+}
```
Comment (mindoodoo) [inline src/shared/systems/serialization.hpp:84]: This seems way more complicated than needed. Why do we even need to have the parentheses ?
Without parentheses we would still be able to serialize entities by using two different separators which would make the code wayyyy more simple.

I strongly feel this is overcomplicated.

File: src/shared/systems/serialization.hpp [hunk 5]
```diff
@@ -0,0 +1,140 @@
...
+
+    std::sregex_iterator words_begin(entity.begin(), entity.end(), componentRegex);
+    std::sregex_iterator words_end = std::sregex_iterator();
+
+    for (std::sregex_iterator i = words_begin; i != words_end; ++i) {
+        components.push_back((*i).str());
+    }
+
+    return components;
+}
```
Comment (mindoodoo) [inline src/shared/systems/serialization.hpp:98]: What is this used for / what does it do ?

**Review by mindoodoo — CHANGES_REQUESTED** [review]

File: src/shared/systems/serialization.hpp (hunk 5 above)
Comment (nLatt) [reply src/shared/systems/serialization.hpp:98]: splitting the entire string into component substrings

File: src/shared/ECS/Components.hpp [hunk 6]
```diff
@@ -0,0 +1,57 @@
...
+*/
+
+#pragma once
+
+#include "ECS.hpp"
+#include "Manager.hpp"
+enum ComponentType { ARMOR, HEALTH, POSITION };
+
+namespace Armor {
+    struct data {
```
Comment (mindoodoo) [inline src/shared/ECS/Components.hpp:15]: Please call this truct "Comp" or "Component" so it reflects the fact it remains the main representation of the component.

Comment (nLatt) [reply src/shared/ECS/Components.hpp]: sure thing

**Review by mindoodoo — APPROVED** [review]
Alles gut

**Pull request merged**